src/thunderbots/software/network_input/filter/particle_filter.h

```diff
@@ -13,17 +14,37 @@ const double MAX_BALL_CONFIDENCE                 = 100.0;
 const double MAX_PARTICLE_STANDARD_DEV           = 0.05;
 const double MIN_PARTICLE_STANDARD_DEV           = 0.001;
 
-const int PARTICLE_FILTER_NUM_CONDENSATIONS = 0.25;
-const double TOP_PERCENTAGE_OF_PARTICLES    = 1.0;
-const double MAX_DETECTION_WEIGHT           = 500.0;
-const double DETECTION_WEIGHT_DECAY         = 2.0;
-const double PREVIOUS_BALL_WEIGHT           = 1.5;
-const double PREDICTION_WEIGHT              = 2.3;
-const double BALL_DIST_THRESHOLD            = 0.5;
-const double BALL_CONFIDENCE_THRESHOLD      = 70.0;
-const double BALL_VALID_DIST_THRESHOLD      = 0.2;
-const double BALL_CONFIDENCE_DELTA          = 6.0;
-const double BALL_MAX_VARIANCE              = 5.0;
+int PARTICLE_FILTER_NUM_CONDENSATIONS =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::
+        PARTICLE_FILTER_NUM_CONDENSATIONS.value();
+double TOP_PERCENTAGE_OF_PARTICLES =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::
+        TOP_PERCENTAGE_OF_PARTICLES.value();
+double MAX_DETECTION_WEIGHT =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::MAX_DETECTION_WEIGHT
+        .value();
+double DETECTION_WEIGHT_DECAY =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::DETECTION_WEIGHT_DECAY
+        .value();
+double PREVIOUS_BALL_WEIGHT =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::PREVIOUS_BALL_WEIGHT
+        .value();
+double PREDICTION_WEIGHT =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::PREDICTION_WEIGHT
+        .value();
+double BALL_DIST_THRESHOLD =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::BALL_DIST_THRESHOLD
+        .value();
+double BALL_CONFIDENCE_THRESHOLD = Util::DynamicParameters::NetworkInput::Filter::
+                                       PartileFilter::BALL_CONFIDENCE_THRESHOLD.value();
+double BALL_VALID_DIST_THRESHOLD = Util::DynamicParameters::NetworkInput::Filter::
+                                       PartileFilter::BALL_VALID_DIST_THRESHOLD.value();
+double BALL_CONFIDENCE_DELTA =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::BALL_CONFIDENCE_DELTA
+        .value();
+double BALL_MAX_VARIANCE =
+    Util::DynamicParameters::NetworkInput::Filter::PartileFilter::BALL_MAX_VARIANCE
+        .value();
```

Comment:
So a dynamic parameter will be updated when you call the `.value()` method. Since these are all in the header, `.value()` will be called only once initially, and these will all take the default value defined in `dynamic_parameters.cpp`. My general pattern for using these is something like the following:
```
void ParticleFilter::update(Point ballPredictedPos)
{
    ...

    // This is the main particle filter loop.
    // - Generate particles around the given basepoints
    // - Update the confidences of these new particles
    // - Evaluate each particle, update their confidence, and keep the top
    // percentage of particles to use
    // as basepoints for the next iteration. These particles should converge to
    // the ball's location.
    int num_condensations = 
        Util::DynamicParameters::NetworkInput::Filter::PartileFilter::
        PARTICLE_FILTER_NUM_CONDENSATIONS.value(); 
    for (int i = 0; i < PARTICLE_FILTER_NUM_CONDENSATIONS; i++)
    {
        // As we loop through each condensation, we want the particles that are
        // generated around the basepoints
        // to get more and more precise, so that we can pinpoint a more accurate
        // point when we take the mean
        
    ...
```
This means that we'll get the new value of the parameter every time the `update` function is called

Reply:
Just a note that `PARTICLE_FILTER_NUM_CONDENSATIONS` should be `num_condensations` here

Reply:
@MathewMacDougall - is your comment here covered by https://github.com/UBC-Thunderbots/Software/pull/413#discussion_r262943787?

Reply:
Yes, but was also pointing out that we're creating the `num_condensations` variable in the example above and then not using it. I assumed it was a typo where `PARTICLE_FILTER_NUM_CONDENSATIONS` was supposed to be `num_condensations`